hawkular-alerts-rest/src/main/java/org/hawkular/alerts/rest/EventsHandler.java

```diff
@@ -97,7 +96,11 @@ public Response createEvent(
                 if (null != alertsService.getEvent(tenantId, event.getId(), true)) {
                     return ResponseUtil.badRequest("Event with ID [" + event.getId() + "] exists.");
                 }
-                alertsService.addEvents(Collections.singletonList(event));
+                /*
+                    New events are sent directly to the engine for inference process.
+                    Input events and new ones generated by the alerts engine are persisted at the end of the process.
+                 */
+                alertsService.sendEvent(event);
```

Comment:
I agree that we should only have one service.  The client should not have to choose, all Events should be persisted and some may go to the engine for evaluation.  It probably doesn't make much difference but I'd probably keep addEvents() as opposed to sendEvents(), because (to me) the primary function is to persist events, and sending them to the engine is something of a side-effect.

The service and engine impls now look good to me with the changes you've made.